chia/data_layer/data_layer_wallet.py

```diff
@@ -78,223 +81,381 @@ async def create_new_dl_wallet(
         cls: Type[_T_DataLayerWallet],
         wallet_state_manager: Any,
         wallet: Wallet,
-        amount: uint64,
-        root_hash: Optional[bytes32],
-        fee: uint64 = uint64(0),
-        name: Optional[str] = None,
-    ) -> ItemAndTransactionRecords[_T_DataLayerWallet]:
+        name: Optional[str] = "DataLayer Wallet",
+    ) -> _T_DataLayerWallet:
         """
         This must be called under the wallet state manager lock
         """
 
         self = cls()
-        self.base_puzzle_program = None
-        self.base_inner_puzzle_hash = None
-        self.standard_wallet = wallet
-        self.log = logging.getLogger(name if name else __name__)
-        std_wallet_id = self.standard_wallet.wallet_id
-        bal = await wallet_state_manager.get_confirmed_balance_for_wallet_already_locked(std_wallet_id)
-        if amount > bal:
-            raise ValueError("Not enough balance")
-        if amount & 1 == 0:
-            raise ValueError("DID amount must be odd number")
         self.wallet_state_manager = wallet_state_manager
-        if root_hash is None:
-            root_hash = Program.to(0).get_tree_hash()
-        self.dl_info = DataLayerInfo(None, root_hash, [], None)
-        info_as_string = json.dumps(self.dl_info.to_json_dict())
-        self.wallet_info = await wallet_state_manager.user_store.create_wallet(
-            "DataLayer Wallet", WalletType.DATA_LAYER.value, info_as_string
-        )
+        self.log = logging.getLogger(name if name else __name__)
+        self.standard_wallet = wallet
+
+        for _, wallet in self.wallet_state_manager.wallets.items():
+            if wallet.type() == uint8(WalletType.DATA_LAYER):
+                raise ValueError("DataLayer Wallet already exists for this key")
+
+        self.wallet_info = await wallet_state_manager.user_store.create_wallet(name, WalletType.DATA_LAYER.value, "")
         if self.wallet_info is None:
             raise ValueError("Internal Error")
-        self.wallet_id = self.wallet_info.id
-        std_wallet_id = self.standard_wallet.wallet_id
-        bal = await wallet_state_manager.get_confirmed_balance_for_wallet_already_locked(std_wallet_id)
-        if amount > bal:
-            raise ValueError("Not enough balance")
+        self.wallet_id = uint8(self.wallet_info.id)
 
+        await self.wallet_state_manager.add_new_wallet(self, self.wallet_info.id)
+
+        return self
+
+    #############
+    # LAUNCHING #
+    #############
+
+    @staticmethod
+    async def match_dl_launcher(launcher_spend: CoinSpend) -> Tuple[bool, Optional[bytes32]]:
```

Comment:
The success value is because quite often you want to do some logic in the case that you _don't_ match a launcher rather than crashing out but you're probably right, `None` would work just fine for that purpose.

Reply:
Also, `except WhateverError:` and `finally:`.